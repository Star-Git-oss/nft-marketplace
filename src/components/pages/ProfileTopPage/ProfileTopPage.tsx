/* eslint-disable @typescript-eslint/no-explicit-any */
import React, { useEffect, useState } from 'react';
import axios from 'axios';
import { useTranslation } from 'react-i18next';
import { useHistory } from 'react-router-dom';
import { Tab, Tabs, TabList, TabPanel } from 'react-tabs';
import Col from '../../common/ui-library/Col/Col';
import Row from '../../common/ui-library/Row/Row';
import Button from '../../common/ui-library/Button/Button';
import NftCard from '../../common/NftCard/NftCard';
import { ApiProxyUri } from '../../../utils/utils';
import 'react-tabs/style/react-tabs.css';

type TempTabContentProps =  {
  nftList: any[];
};

const TempTabContent: React.FC<TempTabContentProps> = ({ nftList }) => (
  <Row>
    {nftList?.map((nft, index) => (
      <NftCard key={nft.id} nft={nft} />
    ))}
    <Col size="20" />
  </Row>
);

const ProfileTopPagePage: React.FC<LoadablePageType> = ({ setIsLoading }) => {

  const { t } = useTranslation();

  const history = useHistory();

  // Nft list in state
  const [nftList, setNftList] = useState([]);

  // // Retrieve NFT info when component loaded
  useEffect(  () => {
    async function fetchData() {
      setIsLoading(true);
      const res = await axios.get(`${ApiProxyUri}/nfts`);
      setNftList(res.data.nfts);
      setIsLoading(false);
    }
    fetchData();
  }, [setIsLoading]);

  return (
    <div>
      <div style={{ textAlign: 'right', margin: '0 auto' }}>
        <Button onClick={() => history.push('/profile')}>Profile</Button>
      </div>
      <Tabs>
        <TabList>
          <Tab>{t('profileTopPage.tab1')}</Tab>
          <Tab>{t('profileTopPage.tab2')}</Tab>
          <Tab>{t('profileTopPage.tab3')}</Tab>
          <Tab>{t('profileTopPage.tab4')}</Tab>
          <Tab>{t('profileTopPage.tab5')}</Tab>
          <Tab>{t('profileTopPage.tab6')}</Tab>
        </TabList>

        <TabPanel>
          <TempTabContent nftList={nftList} />
        </TabPanel>
        <TabPanel>
          <TempTabContent nftList={nftList} />
        </TabPanel>
        <TabPanel>
          <TempTabContent nftList={nftList} />
        </TabPanel>
        <TabPanel>
          <TempTabContent nftList={nftList} />
        </TabPanel>
        <TabPanel>
          <TempTabContent nftList={nftList} />
        </TabPanel>
        <TabPanel>
          <TempTabContent nftList={nftList} />
        </TabPanel>
      </Tabs>
    </div>
  );
};

export default ProfileTopPagePage;