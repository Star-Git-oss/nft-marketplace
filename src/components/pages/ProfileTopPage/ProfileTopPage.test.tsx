/* eslint-disable no-console */
import React from 'react';
import axios from 'axios';
import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import ProfileTopPage from './ProfileTopPage';

// Known issue: error and warning does not work for console
// https://github.com/facebook/react/issues/7047
const original = console.error;

beforeEach(() => {
  console.error = jest.fn();
});

afterEach(() => {
  console.error = original;
});

jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: (key) => key }),
}));

const nftList = {
  nfts: [
    {
      'id':1,
      'labels': [1, 2, 3],
      'name': 'my NFT 1',
      'quantity': '10 of 10',
      'price': '0.2',
      'image': 'https://picsum.photos/200/300',
      'views': 5201,
      'owner': 'mickael canu',
      'creator': 'inri',
      'collectionName': 'Test name'
    },
    {
      'id':2,
      'labels': [1, 2, 3],
      'name': 'my NFT 2',
      'quantity': '10 of 10',
      'price': '0.5',
      'image': 'https://picsum.photos/201/300',
      'views': 5201,
      'owner': 'mickael canu',
      'creator': 'inri',
      'collectionName': 'Test name'
    },
  ]
};

// Axios mock
jest.mock('axios');
const resp = { data: nftList };
axios.get.mockResolvedValue(resp);

// React-router-dom mock
const mockHistoryPush = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    push: mockHistoryPush,
  }),
}));

describe('ProfileTopPage snapshots', () => {

  test('ProfileTopPage renders correctly', () => {
    const tree = renderer
      .create(<ProfileTopPage setIsLoading={()=> jest.fn()} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

});

describe('ProfileTopPage', () => {

  it('renders component tabs properly', () => {
    render(<ProfileTopPage setIsLoading={()=> jest.fn()} />);
    expect(screen.getByText('profileTopPage.tab1')).toBeInTheDocument();
    expect(screen.getByText('profileTopPage.tab2')).toBeInTheDocument();
    expect(screen.getByText('profileTopPage.tab3')).toBeInTheDocument();
    expect(screen.getByText('profileTopPage.tab4')).toBeInTheDocument();
    expect(screen.getByText('profileTopPage.tab5')).toBeInTheDocument();
    expect(screen.getByText('profileTopPage.tab6')).toBeInTheDocument();
  });

  it('Redirect to profile page', () => {
    render(<ProfileTopPage setIsLoading={()=> jest.fn()} />);
    const profile = screen.getByText('Profile');
    expect(profile).toBeInTheDocument();
    fireEvent.click(profile);
    expect(mockHistoryPush).toHaveBeenCalledWith('/profile');
  });

  
});