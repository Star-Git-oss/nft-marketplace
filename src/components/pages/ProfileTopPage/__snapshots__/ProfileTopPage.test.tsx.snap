// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileTopPage snapshots ProfileTopPage renders correctly 1`] = `
<div>
  <div
    style={
      Object {
        "margin": "0 auto",
        "textAlign": "right",
      }
    }
  >
    <button
      className="sc-eCssSg lftEmV"
      onClick={[Function]}
    >
      Profile
    </button>
  </div>
  <div
    className="react-tabs"
    data-tabs={true}
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    <ul
      className="react-tabs__tab-list"
      role="tablist"
    >
      <li
        aria-controls="react-tabs-1"
        aria-disabled="false"
        aria-selected="true"
        className="react-tabs__tab react-tabs__tab--selected"
        id="react-tabs-0"
        role="tab"
        tabIndex="0"
      >
        profileTopPage.tab1
      </li>
      <li
        aria-controls="react-tabs-3"
        aria-disabled="false"
        aria-selected="false"
        className="react-tabs__tab"
        id="react-tabs-2"
        role="tab"
        tabIndex={null}
      >
        profileTopPage.tab2
      </li>
      <li
        aria-controls="react-tabs-5"
        aria-disabled="false"
        aria-selected="false"
        className="react-tabs__tab"
        id="react-tabs-4"
        role="tab"
        tabIndex={null}
      >
        profileTopPage.tab3
      </li>
      <li
        aria-controls="react-tabs-7"
        aria-disabled="false"
        aria-selected="false"
        className="react-tabs__tab"
        id="react-tabs-6"
        role="tab"
        tabIndex={null}
      >
        profileTopPage.tab4
      </li>
      <li
        aria-controls="react-tabs-9"
        aria-disabled="false"
        aria-selected="false"
        className="react-tabs__tab"
        id="react-tabs-8"
        role="tab"
        tabIndex={null}
      >
        profileTopPage.tab5
      </li>
      <li
        aria-controls="react-tabs-11"
        aria-disabled="false"
        aria-selected="false"
        className="react-tabs__tab"
        id="react-tabs-10"
        role="tab"
        tabIndex={null}
      >
        profileTopPage.tab6
      </li>
    </ul>
    <div
      aria-labelledby="react-tabs-0"
      className="react-tabs__tab-panel react-tabs__tab-panel--selected"
      id="react-tabs-1"
      role="tabpanel"
    >
      <div
        className="sc-hKgILt fdNGpG"
      >
        <div
          className="sc-dlfnbm kmkwwG"
          size="20"
        />
      </div>
    </div>
    <div
      aria-labelledby="react-tabs-2"
      className="react-tabs__tab-panel"
      id="react-tabs-3"
      role="tabpanel"
    />
    <div
      aria-labelledby="react-tabs-4"
      className="react-tabs__tab-panel"
      id="react-tabs-5"
      role="tabpanel"
    />
    <div
      aria-labelledby="react-tabs-6"
      className="react-tabs__tab-panel"
      id="react-tabs-7"
      role="tabpanel"
    />
    <div
      aria-labelledby="react-tabs-8"
      className="react-tabs__tab-panel"
      id="react-tabs-9"
      role="tabpanel"
    />
    <div
      aria-labelledby="react-tabs-10"
      className="react-tabs__tab-panel"
      id="react-tabs-11"
      role="tabpanel"
    />
  </div>
</div>
`;
